### ISSUE TITLE
BUG: PIPX Installation wont work (No module named gunicorn)

### ISSUE THREAD

**Issue by b-reich** [Bee-Mar/mmpm#167]
**Describe the Bug**

Is this an issue with the CLI or GUI?
GUI (not 100% sure)
Is this something that should be put in the Wiki?
Yes
mmpm.api logserver and repeater wont work

**To Reproduce**
1. `apt install pipx`
2. `pipx install mmpm`
3. `mmpm guided-setup`
4. `mmpm ui install`

**Expected behavior**
mmpm web ui is startet properly

**Screenshots**

**Operating System, Python3 Version, and Web Browser**\
(Please complete the following information)

- output from `cat /etc/os-release`
```
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/
```
- output from `python3 --version`
`Python 3.11.2`
- Web Browser (firefox, chrome, safari, chromium, etc)

**MMPM Environment Settings**
```json
{
  "MMPM_MAGICMIRROR_ROOT": "/home/pi/MagicMirror",
  "MMPM_MAGICMIRROR_URI": "http://192.168.1.181:8080",
  "MMPM_MAGICMIRROR_PM2_PROCESS_NAME": "MagicMirror",
  "MMPM_MAGICMIRROR_DOCKER_COMPOSE_FILE": "",
  "MMPM_IS_DOCKER_IMAGE": false,
  "MMPM_LOG_LEVEL": "INFO"
}
```
**MMPM Log Files**

- Attach the file produced `Download MMPM log files` within the Control Center of the GUI or from the  `mmpm log --zip`
[mmpm-logs-2024-2-11.zip](https://github.com/Bee-Mar/mmpm/files/14232825/mmpm-logs-2024-2-11.zip)

```
/home/pi/.pm2/logs/mmpm.log-server-error.log last 15 lines:
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn
14|mmpm.lo | /usr/bin/python: No module named gunicorn

/home/pi/.pm2/logs/mmpm.api-error.log last 15 lines:
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn
13|mmpm.ap | /usr/bin/python: No module named gunicorn

/home/pi/.pm2/logs/mmpm.repeater-error.log last 15 lines:
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
15|mmpm.re | /usr/bin/python: No module named gunicorn
```
**MagicMirror Config**

- your MagicMirror `config.js`
 not relevant - can post it when I am wrong

**Comment by Bee-Mar** [comment]
@b-reich I'll add this to the wiki:

```sh
pipx uninstall mmpm
pipx install --include-deps mmpm
```

**Comment by Bee-Mar** [comment]
@b-reich After testing it out, it might require some additional changes in the application to make use of `gunicorn`, etc, in that manner

**Comment by b-reich** [comment]
Think it used the system default python instead of the pipx ones. @Bee-Mar

**Comment by b-reich** [comment]
```
/home/pi/.local/bin is already in PATH.

⚠️  All pipx binary directories have been added to PATH. If you are sure you want to proceed, try again with the '--force' flag.

Otherwise pipx is ready to go! ✨ 🌟 ✨
```

**Comment by b-reich** [comment]
think this is problemematik
https://github.com/Bee-Mar/mmpm/blob/efc79cceec95828cc4d205c4e5a4e4190af3a727/mmpm/ui.py#L47C32-L47C116

**Comment by Bee-Mar** [comment]
> think this is problemematik https://github.com/Bee-Mar/mmpm/blob/efc79cceec95828cc4d205c4e5a4e4190af3a727/mmpm/ui.py#L47C32-L47C116

that's exactly where I'm currently making an adjustment. I'll put out an update soon with that change

**Comment by b-reich** [comment]
> > think this is problemematik https://github.com/Bee-Mar/mmpm/blob/efc79cceec95828cc4d205c4e5a4e4190af3a727/mmpm/ui.py#L47C32-L47C116
> 
> that's exactly where I'm currently making an adjustment. I'll put out an update soon with that change

may `sys.executable` works here https://github.com/Bee-Mar/mmpm/blob/efc79cceec95828cc4d205c4e5a4e4190af3a727/mmpm/ui.py#L31 properly @Bee-Mar

**Comment by b-reich** [comment]
@Bee-Mar If you want me to test your fix, please let me know

**Comment by Bee-Mar** [comment]
Here's an updated wheel I built just now:
[mmpm-4.1.1.tar.gz](https://github.com/Bee-Mar/mmpm/files/14233504/mmpm-4.1.1.tar.gz)

(GitHub makes you upload it as a tar.gz in the issues section)

After downloading it:

```sh
tar xzf mmpm-4.1.1.tar.gz
pip install ./mmpm-4.1.1-py3-none-any.whl
```

**Comment by b-reich** [comment]
@Bee-Mar  works for me. Thanks! When is the pypi release planend?

**Comment by Bee-Mar** [comment]
Great! Within about an hour

**Comment by Bee-Mar** [comment]
@b-reich released `4.1.1` on pypi
